engines/dfc_provider/app/services/dfc_provider/authorization_control.rb

```diff
@@ -8,24 +8,55 @@ def initialize(access_token)
       @access_token = access_token
     end
 
+    def safe_process
+      process
+    rescue JWT::VerificationError, JWT::ExpiredSignature, JWT::ImmatureSignature => e
+      # To help development debugging
+      raise e if Rails.env.development?
+
+      nil
+    end
+
     def process
       decode_token
+      control_payload
       find_ofn_user
     end
 
+    private
+
     def decode_token
       data = JWT.decode(
         @access_token,
         nil,
-        false
+        true,
+        { algorithms: ['RS256'], jwks: jwks_hash }
```

Comment:
Here we are controlling the signature properly. Errors are raised in case of issues.